it/mongodb/src/main/java/org/apache/beam/it/mongodb/DefaultMongoDBResourceManager.java

```diff
@@ -0,0 +1,280 @@
+      String collectionName, boolean createCollection) {
+    if (!collectionExists(collectionName) && !createCollection) {
+      throw new MongoDBResourceManagerException(
+          "Collection " + collectionName + " does not exists in database " + databaseName);
+    }
+
+    return getDatabase().getCollection(collectionName);
+  }
+
+  @Override
+  public synchronized boolean insertDocument(String collectionName, Document document) {
+    return insertDocuments(collectionName, ImmutableList.of(document));
+  }
+
+  @Override
+  public synchronized boolean insertDocuments(String collectionName, List<Document> documents) {
+    LOG.info(
+        "Attempting to write {} documents to {}.{}.",
+        documents.size(),
+        databaseName,
+        collectionName);
+
+    try {
+      getMongoDBCollection(collectionName, /* createCollection= */ true).insertMany(documents);
+    } catch (Exception e) {
+      throw new MongoDBResourceManagerException("Error inserting documents.", e);
+    }
+
+    LOG.info(
+        "Successfully wrote {} documents to {}.{}", documents.size(), databaseName, collectionName);
+
+    return true;
+  }
+
+  @Override
+  public synchronized FindIterable<Document> readCollection(String collectionName) {
+    LOG.info("Reading all documents from {}.{}", databaseName, collectionName);
+
+    FindIterable<Document> documents;
+    try {
+      documents = getMongoDBCollection(collectionName, /* createCollection= */ false).find();
+    } catch (Exception e) {
+      throw new MongoDBResourceManagerException("Error reading collection.", e);
+    }
+
+    LOG.info("Successfully loaded documents from {}.{}", databaseName, collectionName);
+
+    return documents;
+  }
+
+  @Override
+  public synchronized void cleanupAll() {
+    LOG.info("Attempting to cleanup MongoDB manager.");
+
+    boolean producedError = false;
+
+    // First, delete the database if it was not given as a static argument
+    try {
+      if (!usingStaticDatabase) {
+        mongoClient.getDatabase(databaseName).drop();
+      }
+    } catch (Exception e) {
+      LOG.error("Failed to delete MongoDB database {}.", databaseName, e);
+      producedError = true;
+    }
+
+    // Next, try to close the MongoDB client connection
+    try {
+      mongoClient.close();
+    } catch (Exception e) {
+      LOG.error("Failed to delete MongoDB client.", e);
+      producedError = true;
+    }
+
+    // Throw Exception at the end if there were any errors
+    if (producedError) {
+      throw new MongoDBResourceManagerException(
+          "Failed to delete resources. Check above for errors.");
+    }
+
+    super.cleanupAll();
+
+    LOG.info("MongoDB manager successfully cleaned up.");
+  }
+
+  /** Builder for {@link DefaultMongoDBResourceManager}. */
+  public static final class Builder
+      extends TestContainerResourceManager.Builder<DefaultMongoDBResourceManager> {
+
+    @SuppressWarnings("initialization.fields.uninitialized")
```

Comment:
Why? Based on the API here, you should make this nullable, and check for null when you use it.

Reply:
Will address after code drop. Opened https://github.com/apache/beam/issues/27438